### PR TITLE
Feature/augument map

### PR DESCRIPTION
**Pull request by huajian1069**
1. add three choices of map style: standard, silver, night
2. one look-through button on map, for quickly view all nearby favor markers 
3. enforce extracting name from email and store in db when login and sign up, if name is null or empty

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java [hunk 1]
```diff
@@ -352,15 +372,35 @@ private void onToggleClick(View view) {
...
+            CameraUpdateFactory.newLatLngZoom(marker.getPosition(), defaultZoomLevel + 2));
+        return;
+      }
+    }
+    // if all visited
+    for (Marker marker : favorsAroundMe.values()) {
+      ArrayList tagArray = (ArrayList) marker.getTag();
+      tagArray.set(2, false);
+    }
+    Toast.makeText(requireContext(), "you have visited all nearby favors", Toast.LENGTH_SHORT)
```
Comment (mikanikos) [inline app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java:392]: I would extract the string in the strings.xml

File: app/src/main/res/xml/root_preferences.xml [hunk 2]
```diff
@@ -40,6 +40,18 @@
             app:summary="@string/notification_chat_summary"
             app:title="@string/notification_chat_title" />
     </PreferenceCategory>
+
+    <PreferenceCategory android:title="Dispaly">
```
Comment (mikanikos) [inline app/src/main/res/xml/root_preferences.xml:44]: Typo

Comment (huajian1069) [reply app/src/main/res/xml/root_preferences.xml:44]: fixed

File: app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java (hunk 1 above)
Comment (huajian1069) [reply app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java:392]: fixed

**Pull request merged**